### PR TITLE
Remove deprecated use of wait. functions

### PR DESCRIPTION
**Pull request by willdot**
#### Proposed Changes ####

The following functions have been deprecated (along with their replacements)

`PollImmediateUntilWithContext` is now `PollUntilContextCancel`

`PollImmediateInfiniteWithContext` is now `PollUntilContextCancel`

`PollImmediateWithContext` is now `PollUntilContextTimeout`

`PollImmediate` is now `PollUntilContextTimeout`

Since all of these functions that are being replaced as `Immediate` the `immediate` flag is being set as true.

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/9321

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: pkg/secretsencrypt/config.go [hunk 1]
```diff
@@ -238,7 +239,8 @@ func GetEncryptionConfigMetrics(runtime *config.ControlRuntime, initialMetrics b
 
 	// This is wrapped in a poller because on startup no metrics exist. Its only after the encryption config
 	// is modified and the first reload occurs that the metrics are available.
-	err = wait.PollImmediate(5*time.Second, 60*time.Second, func() (bool, error) {
+	ctx := context.Background()
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 60*time.Second, true, func(ctx context.Context) (bool, error) {
 		data, err := restClient.Get().AbsPath("/metrics").DoRaw(context.TODO())
```
Comment (brandond) [inline pkg/secretsencrypt/config.go:244]: nit: use the provided context here?
```suggestion
		data, err := restClient.Get().AbsPath("/metrics").DoRaw(context.TODO())
```

**Review by brandond — APPROVED** [review]
lgtm, one minor nit!

**Comment by willdot** [comment]
> Thank you! lgtm, one minor nit!

Ah good catch. Committed!

**Review by vitorsavian — DISMISSED** [review]

**Review by brandond — APPROVED** [review]

**Comment by dereknola** [comment]
There is an error in the split role cluster config on this PR.

File: pkg/cluster/managed.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func (c *Cluster) setupEtcdProxy(ctx context.Context, etcdProxy etcd.Proxy) {
 		return
 	}
 	// We use Poll here instead of Until because we want to wait the interval before running the function.
-	go wait.PollUntilWithContext(ctx, 30*time.Second, func(ctx context.Context) (bool, error) {
+	go wait.PollUntilContextCancel(ctx, 30*time.Second, true, func(ctx context.Context) (bool, error) {
```
Comment (dereknola) [inline pkg/cluster/managed.go:139]: We explicitly want to wait for the 30s before calling the function. This "may" be the source of the split role error. 
```suggestion
	go wait.PollUntilContextCancel(ctx, 30*time.Second, false, func(ctx context.Context) (bool, error) {
```

**Review by dereknola — CHANGES_REQUESTED** [review]

File: pkg/cluster/managed.go (hunk 2 above)
Comment (willdot) [reply pkg/cluster/managed.go]: Ah I see, thanks for realising that!

🤞  it fixes the tests.

**Review by dereknola — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**